# Data Iris
Kumpulan data bunga Iris adalah kumpulan data multivariat yang diperkenalkan oleh ahli statistik dan biologi Inggris, Ronald Fisher, dalam makalahnya tahun 1936 yang berjudul “The use of multiple measurements in taxonomic problems”. Kumpulan data ini terkadang disebut sebagai kumpulan data Iris Anderson karena Edgar Anderson mengumpulkan data tersebut untuk mengukur variasi morfologi bunga Iris dari tiga spesies yang berkerabat. Kumpulan data ini terdiri dari 50 sampel dari masing-masing tiga spesies Iris (Iris Setosa, Iris virginica, dan Iris versicolor). Empat fitur diukur dari setiap sampel: panjang dan lebar sepal dan petal, dalam sentimeter.  



## Tampilkan data dari CSV

In [1]:
import pandas as pd
df = pd.read_csv('data/IRIS.csv')
df.head(20)

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,Iris-setosa
1,4.9,3.0,1.4,0.2,Iris-setosa
2,4.7,3.2,1.3,0.2,Iris-setosa
3,4.6,3.1,1.5,0.2,Iris-setosa
4,5.0,3.6,1.4,0.2,Iris-setosa
5,5.4,3.9,1.7,0.4,Iris-setosa
6,4.6,3.4,1.4,0.3,Iris-setosa
7,5.0,3.4,1.5,0.2,Iris-setosa
8,4.4,2.9,1.4,0.2,Iris-setosa
9,4.9,3.1,1.5,0.1,Iris-setosa



# Analisis Dataset Iris

## 1. Jumlah Data Tiap Kelas

- Setosa: 50
- Versicolor: 50
- Virginica: 50

Dataset bersifat seimbang (balanced dataset).

---

## 2. Missing Value

| Kolom | Missing Value |
|--------|--------------|
| Sepal Length | 0 |
| Sepal Width | 0 |
| Petal Length | 0 |
| Petal Width | 0 |
| Class | 0 |

Tidak terdapat missing value pada dataset.

---

## 3. Identifikasi Outlier (Metode IQR)

| Fitur | Jumlah Outlier |
|--------|---------------|
| Sepal Length | 0 |
| Sepal Width | 4 |
| Petal Length | 0 |
| Petal Width | 0 |

Outlier hanya ditemukan pada fitur Sepal Width.

---

## 4. Identifikasi Data Aneh (Nilai Minimum & Maksimum)

| Fitur | Min | Max |
|--------|-----|-----|
| Sepal Length | 4.3 | 7.9 |
| Sepal Width | 2.0 | 4.4 |
| Petal Length | 1.0 | 6.9 |
| Petal Width | 0.1 | 2.5 |

Seluruh nilai masih dalam rentang yang wajar.

---

## 5. Rata-Rata Tiap Fitur Berdasarkan Kelas

| Kelas | Sepal Length | Sepal Width | Petal Length | Petal Width |
|--------|-------------|-------------|--------------|-------------|
| Setosa | 5.006 | 3.428 | 1.462 | 0.246 |
| Versicolor | 5.936 | 2.770 | 4.260 | 1.326 |
| Virginica | 6.588 | 2.974 | 5.552 | 2.026 |

Petal Length dan Petal Width merupakan fitur yang paling membedakan antar kelas.

---

## 6. Statistik Deskriptif Keseluruhan

| Statistik | Sepal Length | Sepal Width | Petal Length | Petal Width |
|------------|-------------|-------------|--------------|-------------|
| Mean | 5.843 | 3.057 | 3.758 | 1.199 |
| Std | 0.828 | 0.436 | 1.765 | 0.762 |
| Min | 4.3 | 2.0 | 1.0 | 0.1 |
| 25% | 5.1 | 2.8 | 1.6 | 0.3 |
| 50% | 5.8 | 3.0 | 4.35 | 1.3 |
| 75% | 6.4 | 3.3 | 5.1 | 1.8 |
| Max | 7.9 | 4.4 | 6.9 | 2.5 |

Variasi terbesar terdapat pada Petal Length.

---

## 7. Visualisasi Data

Visualisasi yang digunakan:
- Histogram Sepal Length
- Histogram Petal Length
- Scatter Plot Sepal Length vs Petal Length

Hasil visualisasi menunjukkan:
- Setosa terpisah dengan jelas dari dua kelas lainnya.
- Versicolor dan Virginica memiliki sedikit overlap.
- Terdapat korelasi positif antara Sepal Length dan Petal Length.

---

## Kesimpulan

1. Dataset bersih tanpa missing value.
2. Dataset seimbang antar kelas.
3. Outlier sangat minimal.
4. Fitur paling informatif untuk klasifikasi adalah:
   - Petal Length
   - Petal Width